### PR TITLE
[joltphysics] Properly patching build so that it will install properly

### PR DESCRIPTION
**Pull request by Honeybunch**
I don't know if this issue was ever logged as an issue but the joltphysics package was not working properly. Unfortunately Jolt's build system is quite divergent from vcpkg's. I had to make some invasive removals to remove extra configurations for Distribution / Sanitized targets and made up for it by making sure that the Release build matches the behavior of the Distribution build.

Ideally I can get some of these changes made upstream but until then I wanted to provide this PR as an option for getting a port that's been in vcpkg for a while to actually work. Also I wanted to depend on it and was sad to find that it was non-functional.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Honeybunch** [comment]
Pulled this back to a draft because I just now figured out that I screwed up and trying to link debug will result in an iterator mismatch 😩

**Comment by Honeybunch** [comment]
Oh strange. Only `joltphysics-targets-release.cmake` is generated. No `-debug` version? Weird. If anyone has any ideas I'd love a hand :)

File: ports/joltphysics/fix-export.diff [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ 	# Set linker flags
+ 	set(CMAKE_EXE_LINKER_FLAGS "/SUBSYSTEM:WINDOWS /ignore:4221 /DEBUG:FASTLINK")
+@@ -150,38 +139,11 @@ elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR "${CMAKE_SYSTEM_NAME}" STREQU
+ 	# Set compiler flags for various configurations
+ 	set(CMAKE_CXX_FLAGS_DEBUG "")
+ 	set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+-	set(CMAKE_CXX_FLAGS_DISTRIBUTION "-O3")
+-	set(CMAKE_CXX_FLAGS_RELEASEASAN "-fsanitize=address")
+-	set(CMAKE_CXX_FLAGS_RELEASEUBSAN "-fsanitize=undefined,implicit-conversion,float-divide-by-zero,local-bounds -fno-sanitize-recover=all")
+-	set(CMAKE_CXX_FLAGS_RELEASECOVERAGE "-fprofile-instr-generate -fcoverage-mapping")
```
Comment (dg0yt) [inline ports/joltphysics/fix-export.diff:37]: These removals shouldn't necessary.

However, overriding CMAKE_CXX_FLAGS is a problem. Initial values from the toolchain must be kept. Projects can append additional flags.

File: ports/joltphysics/fix-export.diff [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+-
+-		if (IS_IPO_SUPPORTED)
+-			set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE ON PARENT_SCOPE)
+-			set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_DISTRIBUTION ON PARENT_SCOPE)
+-		else()
+-			message(WARNING "Interprocedural optimizations are not supported: ${IPO_CHECK_OUTPUT}")
+-		endif()
+-	endif()
+-endfunction()
+-SET_INTERPROCEDURAL_OPTIMIZATION()
```
Comment (dg0yt) [inline ports/joltphysics/fix-export.diff:64]: A minimal patch would just remove this line, not the whole function.

File: ports/joltphysics/fix-export.diff [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
++)
++
++install(FILES 
++  joltphysics-config.cmake 
++  DESTINATION share/joltphysics
++)
++
++export(EXPORT joltphysics-targets
++  FILE "${CMAKE_CURRENT_BINARY_DIR}/joltphysics-targets.cmake"
++)
```
Comment (dg0yt) [inline ports/joltphysics/fix-export.diff:99]: Not needed.

**Review by dg0yt — COMMENTED** [review]
When not provided upstream, exported CMake config must be named `unofficial-<PORT>` and use namespace `unofficial::<PORT>::`.
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace

**Comment by Honeybunch** [comment]
> When not provided upstream, exported CMake config must be named `unofficial-<PORT>` and use namespace `unofficial::<PORT>::`. https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace

Oh thanks! I missed this but it should be corrected now

**Comment by Honeybunch** [comment]
Huh, now I'm getting `vcpkg\buildtrees\joltphysics\src\v3.0.1-300ee9d871.clean\Jolt\Core\Memory.cpp.obj: is not a native COFF file. Recompile without /GL?` when I try to link against this :/

**Comment by Honeybunch** [comment]
Had to disable IPO

**Comment by Honeybunch** [comment]
Still getting an issue with the debug target not getting generated though

**Comment by Honeybunch** [comment]
Should have looked at the unofficial-brotli port sooner. Config fixup solved the debug build problems

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**